File: words-api/Lib/BridgeTypes/NounKind.cs
// Project: words-api
// File: NounKind.cs
// 
// Author: Ethan Kisiel (ethan.a.kisiel@gmail.com)
// 
// File Created: 23 08 2025 22:08:10
// Last Modified: 23 08 2025 23:08:20
// 
// Modified By: Ethan Kisiel (ethan.a.kisiel@gmail.com>)
// 
// Copyright 2025 - 2025 Ethan Kisiel, Ethan Kisiel

namespace words_api.Lib.Enums;

public class NounKind
{
    public const string Default = "X";
    public const string Singular = "S";
    public const string Plural = "M";
    public const string AbstractIdea = "A";
    public const string Group = "G";
    public const string Name = "N";
    public const string Person = "P";
    public const string Thing = "T";
    public const string Locale = "L";
    public const string PlaceWhere = "W";

    // X, --  unknown, nondescript
    // S,  --  Singular "only"           --  not really used
    // M,  --  plural or Multiple "only" --  not really used
    // A,  --  Abstract idea
    // G,  --  Group/collective Name -- Roman(s)
    // N,  --  proper Name
    // P,  --  a Person
    // T,  --  a Thing
    // L,  --  Locale, name of country/city
    // W   --  a place Where
}

File: words-frontend/src/lib/models/NounKind.ts
export type NounKindValue = 'X' | 'S' | 'M' | 'A' | 'G' | 'N' | 'P' | 'T' | 'L' | 'W';

export class NounKind {

    private static default = 'X';
    private static singular = 'S';
    private static plural = 'M';
    private static abstractIdea = 'A';
    private static group = 'G';
    private static name = 'N';
    private static person = 'P';
    private static thing = 'T';
    private static locale = 'L';
    private static placeWhere = 'W';

    static getLongForm(value: NounKindValue, detailed=false): string {
        switch (value) {
            case this.default:
                return (!detailed?
                    'Unknown' :
                    'Unknown, nondescript');
            case this.singular:
                return (!detailed?
                    'Singular' :
                    'Singular "only" (not really used)');
            case this.plural:
                return (!detailed?
                    'Plural' :
                    'Plural or Multiple "only" (not really used)');
            case this.abstractIdea:
                return (!detailed?
                    'Abstract Idea' :
                    'Abstract idea');
            case this.group:
                return (!detailed?
                    'Group' :
                    'Group/collective Name — Roman(s)');
            case this.name:
                return (!detailed?
                    'Name' :
                    'Proper name');
            case this.person:
                return (!detailed?
                    'Person' :
                    'A person');
            case this.thing:
                return (!detailed?
                    'Thing' :
                    'A thing');
            case this.locale:
                return (!detailed?
                    'Locale' :
                    'Locale, name of country/city');
            case this.placeWhere:
                return (!detailed?
                    'Place Where' :
                    'A place where');
            default:
                return '';
        }
    }
}